Makefile

```diff
@@ -4,52 +4,88 @@ else
-
 RED=`tput setaf 1`
 GREEN=`tput setaf 2`
+CYAN=`tput setaf 6`
 RESET=`tput sgr0`
 
-START_APP := @$(NPM_BIN)/electron .
-ELECTRON_TEST := ELECTRON_PATH=$(NPM_BIN)/electron $(NPM_BIN)/electron-mocha
-CONFIG := $(THIS_DIR)/desktop/config.json
-DESKTOP_CONFIG := $(THIS_DIR)/desktop-config
-BUILDER := $(THIS_DIR)/builder.js
-BUILD_CONFIG := $(THIS_DIR)/resource/build-scripts/build-config-file.js
-BUILD_DIR := $(THIS_DIR)/build
-PACKAGE_DMG := $(THIS_DIR)/resource/build-scripts/package-dmg.js
-PACKAGE_WIN32 := @$(NPM_BIN)/electron-builder
-CERT_SPC := $(THIS_DIR)/resource/secrets/automattic-code.spc
-CERT_PVK := $(THIS_DIR)/resource/secrets/automattic-code.pvk
 CALYPSO_DIR := $(THIS_DIR)/calypso
-CALYPSO_JS := $(CALYPSO_DIR)/public/build.js
-CALYPSO_CHANGES_STD := `find "$(CALYPSO_DIR)" -newer "$(CALYPSO_JS)" \( -name "*.js" -o -name "*.jsx" -o -name "*.json" -o -name "*.scss" \) -type f -print -quit | grep -v .m. | wc -l`
-CALYPSO_BRANCH = $(shell git --git-dir ./calypso/.git branch | sed -n -e 's/^\* \(.*\)/\1/p')
-WEBPACK_BIN := $(NPM_BIN)/webpack
 
-# sets to 1 if NPM version is >= 3
-NPMGTE3 := $(shell expr `npm -v | cut -f1 -d.` \>= 3)
-MAKENSIS_VERSION := $(shell expr `makensis -VERSION 2>/dev/null`)
+CHECKMARK=✓
+
+CONFIG_ENV = 
+CALYPSO_ENV = desktop
+NODE_ENV = production
+BUILD_PLATFORM = mwl
+
+# Build sources
+# TODO: run tasks parallel when in dev mode
+build-source: checks build-config build-calypso build-desktop
+	@echo "$(CYAN)$(CHECKMARK) All parts built$(RESET)"
+
+# Start app
+start: build-source
+	@echo "$(CYAN)Starting app...$(RESET)"
+
+	@npx electron .
+
+# Build config
+build-config: 
+	@node $(THIS_DIR)/resource/build-scripts/build-config-file.js $(CONFIG_ENV)
+
+	@echo "$(CYAN)$(CHECKMARK) Config created$(RESET)"
+
+# BUild calypso bundle
+build-calypso: 
+	@cd $(CALYPSO_DIR) && NODE_ENV=$(NODE_ENV) CALYPSO_ENV=$(CALYPSO_ENV) npm run -s build
+
+	echo "$(CYAN)$(CHECKMARK) Calypso built$(RESET)"
+
+# Build desktop bundle
+build-desktop:
+	@NODE_ENV=$(NODE_ENV) NODE_PATH=calypso/server$(ENV_PATH_SEP)calypso/client npx webpack --config ./webpack.config.js
+
+	@echo "$(CYAN)$(CHECKMARK) Desktop built$(RESET)"
+
+# Package App
+package:
+	@npx electron-builder build -$(BUILD_PLATFORM)
 
-# check for secrets.json
+	@echo "$(CYAN)$(CHECKMARK) App built$(RESET)"
+
+# Combined steps for building app 
+build: build-source package
+
+# Perform checks
+checks: check-node-and-npm-version-parity secret secret-clientid
+
+# Check for secrets.json
 secret:
-	@if [ ! -f $(THIS_DIR)/calypso/config/secrets.json ]; then if [ -z "${CIRCLECI}" ]; then { echo "calypso/config/secrets.json not found. Required file, see docs/secrets.md"; exit 1; } fi; fi
+	@if [ $(CONFIG_ENV) = "release" ] && [ ! -f $(CALYPSO_DIR)/config/secrets.json ]; \
+	then { \
+		if [ -z "${CIRCLECI}" ]; \
+			then { \
+				echo "$(RED)x calypso/config/secrets.json not found. Required file, see docs/secrets.md$(RESET)"; \
+				exit 1; \
+			} \
+		fi; \
+	} \
+	fi;
```

Comment:
so it looks like we copied a BASH script into Make without transforming the paradigm. just pointing this out even though it was probably existing…

```Makefile
calypso/config/secrets.json:
ifndef $(CIRCLECI)
	$(info $(RED)x $@ not found. Required file, see docs/secrets.md$(RESET))
else
endif
```

^^^ doing that we can then incorporate it into some `make release` command or something like it.

this is a random comment, please don't feel pressured